tests/environment.test.js

```diff
@@ -92,8 +97,164 @@ tap.test('Test Environment variables', test => {
+      assert.strictSame(error.message, errorMessage)
+    }
+
+    assert.end()
+  })
+
+  test.test('Should fail since CONDITION_FIELD is false and CONDITION_FALSE_REQUIRED_FIELD is not present', async assert => {
+    const env = {
+      ...baseEnv,
+      CONDITION_FIELD: false,
+    }
+
+    // NOTE: use try catch instead of assert.reject to customize error message assertion
+    assert.plan(1)
+    try {
+      await setupFastify('./tests/services/if-then-else-env-validation-custom-service.js', env)
+    } catch (error) {
+      const errorMessage = 'env must have required property \'CONDITION_FALSE_REQUIRED_FIELD\', env must match "else" schema, env must have required property \'BASE_REQUIRED_FIELD\''
+      assert.strictSame(error.message, errorMessage)
+    }
+
+    assert.end()
+  })
+
+  test.test('Should pass since CONDITION_FIELD is true and CONDITION_FALSE_REQUIRED_FIELD is present', async assert => {
+    const env = {
+      ...baseEnv,
+      BASE_REQUIRED_FIELD: 'some-value',
+      CONDITION_FIELD: true,
+      CONDITION_TRUE_REQUIRED_FIELD: 'some-value',
+    }
+
+    await assert.resolves(async() => {
+      await setupFastify('./tests/services/if-then-else-env-validation-custom-service.js', env)
+    })
+
+    assert.end()
+  })
+
+  test.test('Should fail since none of the anyOf required fields are present', async assert => {
+    // NOTE: use try catch instead of assert.reject to customize error message assertion
+    assert.plan(1)
+    try {
+      await setupFastify('./tests/services/any-of-env-validation-custom-service.js', baseEnv)
+    } catch (error) {
+      const errorMessage = 'env must have required property \'ANY_OF_REQUIRED_FIELD_1\', env must have required property \'ANY_OF_REQUIRED_FIELD_2\', env must match a schema in anyOf'
+      assert.strictSame(error.message, errorMessage)
+    }
+
+    assert.end()
+  })
+
+  test.test('Should pass since one of the anyOf required fields is present', async assert => {
+    const env = {
+      ...baseEnv,
+      ANY_OF_REQUIRED_FIELD_1: 'some-value',
+    }
+
+    await assert.resolves(async() => {
+      await setupFastify('./tests/services/any-of-env-validation-custom-service.js', env)
+    })
+
+    assert.end()
+  })
+
+  test.test('Should fail since not all of the allOf required fields are present', async assert => {
+    const env = {
+      ...baseEnv,
+      ALL_OF_REQUIRED_FIELD_1: 'some-value',
+    }
+
+    // NOTE: use try catch instead of assert.reject to customize error message assertion
+    assert.plan(1)
+    try {
+      await setupFastify('./tests/services/all-of-env-validation-custom-service.js', env)
+    } catch (error) {
+      const errorMessage = 'env must have required property \'ALL_OF_REQUIRED_FIELD_2\''
+      assert.strictSame(error.message, errorMessage)
+    }
+
+    assert.end()
```

Comment:
Is it correct to use plan with end?

Reply:
You're absolutely right, it is also not necessary if the test callback returns a promise so I removed all the `assert.end()` of the test file, since all the tests are `async`